tachyon_core/src/storage/writer/persistent_writer.rs

```diff
@@ -47,18 +47,21 @@ impl PersistentWriter {
             .partial_init(ts, v)
         } else {
             let file_path = PersistentWriter::derive_file_path(&self.root, stream_id, ts);
-            self.indexer
-                .borrow_mut()
-                .insert_new_file(stream_id, &file_path, ts, None)
-                .unwrap();
 
-            PartiallyPersistentDataFile::new(
+            let file = PartiallyPersistentDataFile::new(
                 self.version,
                 StreamId(stream_id.as_u128()),
                 value_type,
                 file_path.clone(),
             )
-            .lazy_init(ts, v)
+            .lazy_init(ts, v);
+
+            self.indexer
+                .borrow_mut()
+                .insert_new_file(stream_id, &file_path, ts, None)
+                .unwrap();
```

Comment:
Same here? Might have to add new errors types on Writer

Reply:
Was able to use the error types from https://github.com/tachyontsdb/tachyon/pull/70